src/v8/v8_context.rs

```diff
@@ -50,26 +51,33 @@ impl V8Context {
         }
     }
 
-    /// Set a private data on the context that can later be retieve with `get_private_data`.
+    /// Set a private data on the context that can later be retieved with `get_private_data`.
     pub fn set_private_data<T>(&self, index: usize, pd: Option<&T>) {
```

Comment:
Should this return some guard that will unset it? and make sure the value set is not outlive the pd lifetime?

Reply:
Also do we still need it to be an Option?